core/src/main/scala/org/http4s/Query.scala

```diff
@@ -106,14 +106,43 @@ object Query {
 
   val empty: Query = new Query(Vector.empty)
 
+  final class QueryEncodeParams private (
```

Comment:
Why are only these four special?  [This post](https://bugzilla.mozilla.org/show_bug.cgi?id=1152455#c6) demonstrates how widely browsers disagree.

My inclination was that we could take a `Char => Boolean`.  But that would permit all sorts of ill-advised renderings as well.